$NetBSD: patch-native_fdlibm_ieeefp_h,v 1.3 2021/12/27 03:31:57 dholland Exp $

Build on AArch64 (64-bit ARM).

Fix NetBSD PPC build, from Magnus Henoch in PR 34787.

--- native/fdlibm/ieeefp.h.orig	2005-12-07 17:54:35.000000000 +0000
+++ native/fdlibm/ieeefp.h
@@ -27,6 +27,15 @@
 #endif
 #endif
 
+#if defined(__aarch64__) || defined(__AARCH64__)
+/* 64-bit ARM */
+#ifdef __ARM_BIG_ENDIAN 1
+#define __IEEE_BIG_ENDIAN
+#else
+#define __IEEE_LITTLE_ENDIAN
+#endif
+#endif
+
 #ifdef __hppa__
 #define __IEEE_BIG_ENDIAN
 #endif
@@ -123,9 +132,9 @@
 #define __IEEE_LITTLE_ENDIAN
 #endif
 
-#if defined (__PPC__) || defined (__ppc__) || defined (__ppc64__) || defined (_POWER)
+#if defined (__PPC__) || defined (__ppc__) || defined (__ppc64__) || defined (_POWER) || defined(__powerpc__)
 #if (defined(_BIG_ENDIAN) && _BIG_ENDIAN) || (defined(_AIX) && _AIX) \
-    || defined (__APPLE__)
+    || defined (__APPLE__) || defined(__NetBSD__)
 #define __IEEE_BIG_ENDIAN
 #else
 #if (defined(_LITTLE_ENDIAN) && _LITTLE_ENDIAN) || (defined(__sun__) && __sun__) || (defined(__WIN32__) && __WIN32__)
